### DISCUSSION TITLE
Fork Hedley

### DISCUSSION THREAD

**Discussion by falbrechtskirchinger** [nlohmann/json#3551] [Ideas]
Hedley's development seems to have stagnated [1] and we've added some macros recently that Hedley should be providing. Additionally, Hedley doesn't offer generic warning suppression macros like doctest from which the library code would benefit. 

Should we consider maintaining a fork of Hedley, adding things like `JSON_NO_UNIQUE_ADDRESS` and doctest-like warning suppressions?

The objective of the fork would be to maintain additions useful to this library until upstream activity picks up again, and not to be an independent project. I'd also be willing to maintain this fork myself if that would help, and try to get changes upstreamed.

Edit: An example of the kind of code I'd like to clean up:
```cpp
#ifdef __GNUC__
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wfloat-equal"
#endif
        const_reference lhs = *this;
        JSON_IMPLEMENT_OPERATOR( ==, true, false, false)
#ifdef __GNUC__
#pragma GCC diagnostic pop
#endif
```

```cpp
JSON_HEDLEY_GCC_SUPPRESS_WARNING_PUSH
JSON_HEDLEY_GCC_SUPPRESS_WARNING("-Wfloat-equal")
        const_reference lhs = *this;
        JSON_IMPLEMENT_OPERATOR( ==, true, false, false)
JSON_HEDLEY_GCC_SUPPRESS_WARNING_POP
```

[1] For example, `[[no_unique_address]]` was discussed but nothing has materialized so far.

**Comment by nlohmann** [comment]
Looking into it.

---
*This reply was drafted by [Claude Code](https://claude.com/claude-code) on behalf of @nlohmann, as part of a review of open discussions.*
